File: palindromy.cpp
/*
 * palindromy.cpp
 * 
 * 
 */


#include <iostream>
using namespace std;

int zlicz_znaki(char t[]) {
	int i = 0;
	int r = 0;
	while (t[i] != '\0') {
		i++;
	} 
	return i; // zwrocenie rozmiaru tekstu
}


int czy_palindrom(wyraz[]) {
	int r = zlicz_znaki(wyraz); // długość wyrazu
	bool czy_p = true;
	for (int i = 0; i < r/2; i++) {
		if (wyraz[i] != wyraz[r - 1 - i]) }
			czy_p = false;
			break;
			//return false;
		}
	}
	
	return czy_p;
	//return true;
	
int main(int argc, char **argv)
{
	int r = 10;
	char wyraz[r]
	cout << "Podaj wyraz: ";
	cin.getline(wyraz, r);
	
	if (czy_palindrom(wyraz)) cout << "To palindrom!";
	else cout << "To nie jest palindrom";
	cout << endl;
	
	return 0;
}

File: palindromy.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  palindromy.py
#  


def main(args):
	
	a = input("Podaj słowo: ")
	i = 0
	b = len(w) - 1
	jest = True
	while(i < j and jest):
		if(a[i] != a[b]):
			jest = False
		i = i + 1
		b = b - 1
	if(jest):
		print("To jest palindrom!")
	else: 
		print("To nie jest palindrom!")
	
	
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
